ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -259,7 +259,9 @@ private module Cached {
       exists(any(Call c).getKeywordArgument(name))
       or
       FlowSummaryImplSpecific::ParsePositions::isParsedKeywordParameterPosition(_, name)
-    }
+    } or
+    TAnyArgumentPosition() or
+    TAnyKeywordArgumentPosition()
```

Comment:
I am not sure why we need this, actually?

Reply:
In https://github.com/github/codeql/pull/9215/commits/0879b6ae12a48c28ee27270fdd32c2ff0cf0ec30#diff-3a6f42ed2c69a87189b2f6c85dae0a2162031d5b4371d02b9d0770b446c25423 i make it such that Argument positions are used for `Argument[...]` parsing for path-components in MaD, and Parameter positions are used fro `Parameter[...]` parsing.

Since we want to allow you to write both `Argument[any/any-named]` and `Parameter[any/any-named]`, we also need both any/any-named positions for both parameters and arguments